api/src/main/java/org/openmrs/module/ehrreports/reporting/library/cohorts/Moh717CohortQueries.java

```diff
@@ -80,4 +80,22 @@ public CohortDefinition getPatientStates(EhrReportConstants.OccurenceStates stat
     cd.addCalculationParameter("state", state);
     return cd;
   }
+ public CohortDefinition getMchClients(int answer) {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Mch Clients");
+    cd.addParameter(new Parameter("startDate", "StartDate", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.setQuery(
+            Moh717Queries.getMchClients(
+                    outpatientMetadata.getAdultsInitialEncounterType().getEncounterTypeId(),
+                    outpatientMetadata.getAdultReturnEncounterType().getEncounterTypeId(),
+                    outpatientMetadata.getPedsInitialEncounterType().getEncounterTypeId(),
+                    outpatientMetadata.getPedsReturnEncounterType().getEncounterTypeId(),
```

Comment:
For this case, you can concentarate on the concepts questions of triage/special clinic and have value coded to MCH clinic, you can do away with encounters for now

Reply:
How are you applying the PNC, ANC, CWC etc, how are those related to the MCH query?